### PR TITLE
updates to createTreeFromFileMap that support sync deletes.

### PR DESCRIPTION
**Pull request by carterm**
This originally was the PR for the Wordpress-to-github project.  Which has been moved here...
https://github.com/cagov/wordpress-to-github

Reduced to a single library update for the tree creation function.

Includes update to `createTreeFromFileMap` that supports deletes and direct commits.

### CODE REVIEW

File: WordpressSync/endpoints.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          "PostPath": "wordpress/posts",
+          "PagePath": "wordpress/pages",
+          "SyncMedia": true,
+          "MediaPath": "wordpress/media"
+        }
+      },
+      {
+        "name": "Cannabis",
+        "enabled": false,
+        "WordPressUrl": "https://dev-cagov-dcc.pantheonsite.io",
```
Comment (aaronhans) [inline WordpressSync/endpoints.json:38]: production url is currently: https://staginginye.prod3.sites.ca.gov/

Comment (aaronhans) [reply WordpressSync/endpoints.json:38]: will change to cannabis.ca.gov on 7/1

**Comment by jbum** [comment]
This is pretty ancient, should we close it?

**Comment by carterm** [comment]
All of this has been incorporated into github-tree-push, which we can update later.

**Pull request closed without merging**